### DISCUSSION TITLE
Errors while using Example Applauncher

### DISCUSSION THREAD

**Discussion by inkeaton** [Aylur/ags#331] [Q&A] [answered]
Hi!
I've been using ags for a while for my top bar, and i was thinking about using it to substitute my app launcher, rofi.
I've downloaded the example widget, and when i try to run it, i get the following error/warning:

```
(com.github.Aylur.ags:5779): Gjs-WARNING **: 16:52:31.571: JS ERROR: TypeError: this is undefined
query@resource:///com/github/Aylur/ags/service/applications.js:70:9
Applauncher@file:///home/inkeaton/.config/ags/samarrun/applauncher.js:30:29
@file:///home/inkeaton/.config/ags/samarrun/applauncher.js:100:12
async*_init/GLib.MainLoop.prototype.runAsync/</<@resource:///org/gnome/gjs/modules/core/overrides/GLib.js:266:34
```

The widget is not shown, and ags does not crash. I have no idea on what may be the cause of the error. 
I am on NixOs, unstable branch.

Do you have any idea? thanks in advance

**Comment by Aylur** [comment, marked as answer]
I'm guessing your ags version is a bit out of date. In javascript class methods are not bound to instances, but in 1.7.5 they were converted into fat arrow functions
```js
// pre 1.7.5, query function loses its `this` context
const { query } = await Service.import('applications')

// solve this by
const apps = await Service.import('applications')
const query = apps.query.bind(apps)
```
or update ags

**Reply by inkeaton** [reply]
Thank you for your quick answer ( and since i'm already here, thank you for the creation of this tool :) )

I've tried your fix, and it fixes the immediate error, the widget still does not show up. Maybe i simply need to update.
I'm on NixOS, and i've installed ags via the home-manager module, as [described here](https://aylur.github.io/ags-docs/config/home-manager/). I've already rebuilt, and still, it does not seem to have been updated. May i ask you if you know how should i update? Should i build it myself?

Thanks in advance for your answer, and i hope to not be bothering you :D

**Reply by Aylur** [reply]
it doesn't shop up by default, you have to run `ags -t applauncher`

**Reply by inkeaton** [reply]
OH, ok. Thank you, sorry for the late response,
